lib/oc-tests/test_dirDepthDownload.py

```diff
@@ -0,0 +1,173 @@
+    d = make_workdir()
+    uploader_num = get_user_number_from_work_directory(d)
+
+    step (3,'Create directories and files then sync and validate')
+
+    filename = "%s%s" % (d,'/TEST_FILE_NEW_USER_SHARE.dat')
+    createfile(filename,'0',count=1000,bs=filesizeKB)
+    upload_dir = "%s%d" % ('uploader', uploader_num)
+
+    procName = reflection.getProcessName()
+
+    if style is 'flat':
+        for i in range(dir_depth):
+            upload_dir = "%s_%d"%('upload_dir', i) 
+            webdav_mkcol (upload_dir, user_num = uploader_num)
+            for j in range(0, numFilesToCreate):
+                upload_name = "%s_%d.dat" % ('TEST_FILE_NEW_USER_SHARE', j)
+                webdav_upload(filename, upload_dir, upload_name, uploader_num)
+    else:
+        upload_dir = procName
+        webdav_mkcol (upload_dir, user_num = uploader_num)
+        for i in range(dir_depth):
+            upload_dir = "%s/%s_%d"%(upload_dir, 'upload_dir', i)
+            webdav_mkcol (upload_dir, user_num = uploader_num)
+            for j in range(0, numFilesToCreate):
+                upload_name = "%s_%d.dat" % ('TEST_FILE_NEW_USER_SHARE', j)
+                webdav_upload(filename, upload_dir, upload_name, uploader_num)
+
+    delete_file (filename)
+
+    run_ocsync(d,user_num=uploader_num)
+
+    list_files(d, recursive=True)
+
+    if style is 'flat':
+        checkFilesExist(d) 
+    else:
+        dirToCheck = "%s/%s" %(d,procName)
+        checkFilesExist(dirToCheck) 
+
+    step (4, 'Uploader final step')
+
+for i in range(num_users):
+    add_worker (uploader,name="uploader%02d"%(i+1))
+
+def checkFilesExist (work_dir):
+
+    logger.info ('Checking if files exist in local directory %s' % work_dir)
+
+    if style is 'flat':
+        for i in range(dir_depth):
+            dir_name = "%s/%s_%d"%(work_dir, 'upload_dir', i)
+            for j in range(0, numFilesToCreate):
+                filename = "%s%s%i%s" % (dir_name, '/TEST_FILE_NEW_USER_SHARE_',j,'.dat')
+                full_name = os.path.join(work_dir, filename)
+                logger.info ('Checking that %s is present ', full_name)
+                expect_exists (full_name)
+    else:
+        dir_name = work_dir
+        for i in range(dir_depth):
+            dir_name = "%s/%s_%d"%(dir_name, 'upload_dir', i)
+            for j in range(0, numFilesToCreate):
+                filename = "%s%s%i%s" % (dir_name, '/TEST_FILE_NEW_USER_SHARE_',j,'.dat')
+                full_name = os.path.join(work_dir, filename)
+                logger.info ('Checking that %s is present ', full_name)
+                expect_exists (full_name)
+
+
+def get_user_number_from_work_directory(dir):
+    """
+    :param dir: string Path of the directory
+        /home/user/smashdir/test_uploadFiles-150522-111229/shareeTwo01
+    :return: integer User number from the last directory name
+    """
+
+    dir = dir[len(config.rundir) + 1:]
+    user_num = int(re.search(r'\d+', dir).group())
```

Comment:
As long as we have the directory name under control, there is no need to change this, but this is a potential cause of error in the test. Directory names such as "fo_4_4_5" will make the test crash; this shouldn't be the case, but we should be aware of this